editor/animation_track_editor.cpp

```diff
@@ -4240,7 +4240,8 @@ AnimationTrackEditor::TrackIndices AnimationTrackEditor::_confirm_insert(InsertD
 				PropertyInfo h = _find_hint_for_track(animation->get_track_count() - 1, np);
 				animation->remove_track(animation->get_track_count() - 1); // Hack.
 
-				if (h.type == Variant::FLOAT ||
+				if (h.type == Variant::INT ||
```

Comment:
Vector2i, Vector3i and Rect2i should probably be added here too, as they use integer components.

Reply:
Done.